src/BSON/Timestamp.c

```diff
@@ -67,6 +67,48 @@ static bool php_phongo_timestamp_init(php_phongo_timestamp_t *intern, phongo_lon
 	return true;
 }
 
+/* Initialize the object from numeric strings and return whether it was
+ * successful. An exception will be thrown on error. */
+static bool php_phongo_timestamp_init_from_string(php_phongo_timestamp_t *intern, const char *s_increment, phongo_zpp_char_len s_increment_len, const char *s_timestamp, phongo_zpp_char_len s_timestamp_len TSRMLS_DC)
+{
+	int64_t increment, timestamp;
+	char *endptr = NULL;
+
+	errno = 0;
+
+	/* errno will set errno if conversion fails; however, we do not need to
+	 * specify the type of error.
+	 *
+	 * Note: bson_ascii_strtoll() does not properly detect out-of-range values
+	 * (see: CDRIVER-1377). strtoll() would be preferable, but it is not
+	 * available on all platforms (e.g. HP-UX), and atoll() provides no error
+	 * reporting at all. */
+
+#if defined(PHP_WIN32)
+	increment = _atoi64(s_increment);
+#else
+	increment = bson_ascii_strtoll(s_increment, &endptr, 10);
+#endif
+
+	if (errno || (endptr && endptr != ((const char *)s_increment + s_increment_len))) {
+		phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Error parsing \"%s\" as 64-bit integer for %s initialization", s_increment, ZSTR_VAL(php_phongo_utcdatetime_ce->name));
```

Comment:
It's towards the end of this line. You're currently reporting the UTCDateTime class name instead of Timestamp.